### PR TITLE
Desktop: Resolves #9193: Upgrade CodeMirror dependencies

### LINKED ISSUE #9193: URL syntax highlighting is incorrect on certain URLs
In the beta editor, create a link with subdomain, such as `https://en.m.wikipedia.org/wiki/Duck`

`https://en.m.wikipedia` will be highlighted in blue, while the rest of the link will be in black. In practice, the whole link should be blue.

image

This applies to both mobile and desktop

### PR DESCRIPTION
**Pull request by personalizedrefrigerator**
# Summary

- Updates CodeMirror dependencies in `packages/editor`
- Removes an unused CodeMirror dependency of `packages/app-mobile`

Resolves #9193.

# Testing

Desktop:
0. Enable the beta editor
1. Enable Vim mode
2. Verify that `/` find works
3. Move with `h`/`j`/`k`/`l`
4. Disable Vim mode
5. Paste a screenshot into the editor
6. Drag and drop a note into the editor
7. Paste `https://en.m.wikipedia.org/wiki/Duck` into the editor (verify it highlights as a link)

Tested on Ubuntu 23.10.

Mobile:
1. Write `This is a test...`
2. Select `test` and click "bold"
3. Press the indent button to give the just-written text code formatting.
4. Create a three-item numbered list with the "toggle list" button
5. Indent the second item
6. Verify that the list re-numbered
8. De-indent the second item

Tested on Android 7.

### CODE REVIEW

**Comment by laurent22** [comment]
There are few conflicts after merging other PRs

**Comment by personalizedrefrigerator** [comment]
With updates applied, the recently-added list-renumbering test is failing (though it seems to work when I try something equivalent to the test manually).

I'm now trying to determine why. **Edit**: It seems that perhaps part of CodeMirror's syntax tree isn't loading &mdash; it's only recognizing one of two lists as containing a cursor...

File: packages/editor/CodeMirror/markdown/markdownReformatter.test.ts [hunk 1]
```diff
@@ -144,34 +144,24 @@ describe('markdownReformatter', () => {
 		expect(tabsToSpaces(state, '  \t  ')).toBe('      ');
 	});
 
-	it('should correctly renumber a list with multiple selections', async () => {
-		const firstListText = [
+	it('should correctly renumber a list with multiple selections in that list', async () => {
+		const listText = [
```
Comment (personalizedrefrigerator) [inline packages/editor/CodeMirror/markdown/markdownReformatter.test.ts:148]: The original version of this test fails with the new version of CodeMirror. Testing list renumbering with multiple cursors manually, however, works.

I suspect that the CodeMirror syntax tree for ordered lists isn't loading correctly (or isn't being forced to load).

**Comment by laurent22** [comment]
What is the status of the is PR? Is there more to do?

**Comment by personalizedrefrigerator** [comment]
> What is the status of the is PR? Is there more to do?

It's ready! I'll fix the merge conflicts.

**Comment by personalizedrefrigerator** [comment]
Never mind! It's no longer relevant (https://github.com/laurent22/joplin/pull/9563 was merged, which involved upgrading CodeMirror to avoid duplicate dependencies).

**Pull request closed without merging**